content/src/service/access/AccessCheckerForWearables.ts

```diff
@@ -100,6 +152,19 @@ export class AccessCheckerForWearables {
     }
   }
 
+  private async calculateHashes(content?: Map<string, string>, metadata?: any) {
+    // Compare both by key and hash
+    const compare = (a: { key: string; hash: string }, b: { key: string; hash: string }) => {
+      if (a.hash > b.hash) return 1
+      else if (a.hash < b.hash) return -1
+      else return a.key > b.key ? 1 : -1
+    }
+    const entries = Array.from(content?.entries() ?? [])
+    const contentAsJson = entries.map(([key, hash]) => ({ key, hash })).sort(compare)
+    const buffer = Buffer.from(JSON.stringify({ content: contentAsJson, metadata }))
+    return Promise.all([Hashing.calculateBufferHash(buffer), Hashing.calculateIPFSHash(buffer)])
```

Comment:
@menduz if we are changing the `calculateIPFSHash` here, how will it impact the contract?

Reply:
I think we should only use the new hashes. Here is a pertinent PR https://github.com/decentraland/catalyst-commons/pull/127